File: leetcode1590/solveLeetcode1590.go
// func minSubarray(nums []int, p int) int {

// 	if p == 1 {

// 		return 0

// 	}

// 	n := len(nums)

// 	sum := 0

// 	for _, num := range nums {

// 		sum = (sum + num) % p

// 	}

// 	prefix := 0

// 	m := make(map[int]int)

// 	m[0] = -1

// 	ans, t := n, 0

// 	for i := 0; i < n; i++ {

// 		prefix = (prefix + nums[i]) % p
// 		m[prefix] = i
// 		t = (prefix - sum + p) % p

// 		if val, ok := m[t]; ok {

// 			ans = min(ans, i-val)

// 		}

// 	}

// 	if ans < n {

// 		return ans

// 	}

// 	return -1

// }

File: leetcode1590/solveLeetcode1590.cpp
// class Solution 
// {
// public:
//     int minSubarray(vector<int>& nums, int p) 
//     {
//         if(p==1)return 0;
//         const int n=nums.size();

//         int sum=accumulate(nums.cbegin(),nums.cend(),0LL)%p;
//         //sum%=p;
        
//         int prefix=0;

//         unordered_map<int,int>hash{{0,-1}};
//         //hash[0]=1;

//         int ans=n,t;
//         for(int i=0;i<n;++i)
//         {
//             prefix=(prefix+nums[i])%p;
//             hash[prefix]=i;

//             t=(prefix-sum+p)%p;
//             if(hash.count(t))
//                 ans=min(ans,i-hash[t]);   
//         }

//         return ans<n?ans:-1;
//     }   
// };